### PR TITLE
user rate book

### PR DESCRIPTION
**Pull request by PhanTrungDuc**

### CODE REVIEW

File: app/views/ratings/_rating.html.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<%= t"views.ratings.rating" %>:
+<%= @rate.round Settings.ratings.round %>/<%= Settings.ratings.max_rate %>
+(<%= pluralize @book.ratings.count, t("views.ratings.vote") %>)
```
Comment (yeunuthongminh) [inline app/views/ratings/_rating.html.erb]: end file

File: app/models/book.rb [hunk 2]
```diff
@@ -15,4 +16,8 @@ class Book < ApplicationRecord
 
   scope :order_desc, -> {order created_at: :DESC}
   scope :publish_date_order_desc, ->{order publish_date: :DESC}
+
+  def not_rated? user
+    return self.ratings.find_by(user: user) ? false : true
```
Comment (yeunuthongminh) [inline app/models/book.rb]: self.ratings.find_by user: user

File: app/models/user.rb [hunk 3]
```diff
@@ -11,7 +11,8 @@ class User < ApplicationRecord
     foreign_key: "followed_id", dependent: :destroy
   has_many :following, through: :active_relationships, source: :followed
   has_many :followers, through: :passive_relationships, source: :follower
- 
+  has_many :ratings
+  
   attr_accessor :remember_token
   before_save {self.email = email.downcase}
```
Comment (yeunuthongminh) [inline app/models/user.rb]: roi viet cai trong ngoac thanh ham
before_save :<ten ham>

File: app/assets/stylesheets/custom.scss [hunk 4]
```diff
@@ -327,6 +327,14 @@ input {
...
+
+.rating {
+    padding: 5px 0;
+  }
+  .quantity {
+    padding: 5px 0 10px 0;
+}
+
 /* review */
 .div-review{
```
Comment (yeunuthongminh) [inline app/assets/stylesheets/custom.scss]: .div-review {

File: app/views/books/show.html.erb [hunk 5]
```diff
@@ -2,20 +2,35 @@
...
             <span class="glyphicon glyphicon-heart" aria-hidden="true"/>
           <% end %>
         </div>
+      <div class="row">
+      </div>  
+        <div class="rating"><%= t "views.ratings.rating" %>:
+          <%= @rate.round Settings.ratings.round %>/<%= Settings.ratings.max_rate %>
+          (<%= pluralize @book.ratings.count, t("views.ratings.vote") %>)
+        </div>
+        <% if logged_in? && @book.not_rated?(current_user)  %>
```
Comment (yeunuthongminh) [inline app/views/books/show.html.erb]: indent

File: app/assets/stylesheets/custom.scss [hunk 6]
```diff
@@ -327,8 +327,16 @@ input {
 .avartar{
   margin-top: 20px
 }
+
+.rating {
+    padding: 5px 0;
+  }
+  .quantity {
```
Comment (yeunuthongminh) [inline app/assets/stylesheets/custom.scss]: indent

File: app/assets/stylesheets/custom.scss [hunk 7]
```diff
@@ -327,8 +327,16 @@ input {
 .avartar{
   margin-top: 20px
 }
+
+.rating {
+    padding: 5px 0;
+  }
```
Comment (yeunuthongminh) [inline app/assets/stylesheets/custom.scss]: xuong dong

**Comment by yeunuthongminh** [comment]
OK

File: app/controllers/ratings_controller.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
+class RatingsController < ApplicationController
+  before_action :logged_in_user
+  before_action :find_book
+
+  def create
+    @rating = current_user.ratings.build book: @book,
+      rate: rating_params[:rate]
```
Comment (hellovietnam93) [inline app/controllers/ratings_controller.rb]: wtf, sao k dùng kiểu tạo param rồi gán biến

File: app/models/book.rb [hunk 9]
```diff
@@ -15,4 +16,8 @@ class Book < ApplicationRecord
 
   scope :order_desc, -> {order created_at: :DESC}
   scope :publish_date_order_desc, ->{order publish_date: :DESC}
+
+  def not_rated? user
```
Comment (hellovietnam93) [inline app/models/book.rb]: rated

**Pull request merged**